az-digital/az_quickstart
# Closes #2528 Add bootstrap class `col-form-label` to `<legend>` element within `<fieldset>`s.

## Description

Within the PR above you will see a new template for az_barrio that handles fieldsets.


Bootstrap's opinion on this: https://getbootstrap.com/docs/4.0/components/forms/#horizontal-form-label-sizing

The only change to the [existing template](https://git.drupalcode.org/project/bootstrap_barrio/-/blob/5.x/templates/form/fieldset.html.twig) is the addition of the `col-form-label` class to the  `legend` element.

```
{%
  set legend_classes = [
    'col-form-label',
   'font-weight-bold',
  ]
%}
```

Extensible
```
  <legend{{ legend.attributes.addClass(legend_classes) }}>
```

## Related issues
- Closes #2528 
- ~~Blocked by #3279 (blocked because it is difficult to test without demo content.)~~

## How to test

Go to [`/finders/events`](https://9a355b9d-e15c-41ca-a57e-f8b221c9a044.probo.build/finders/events)
Notice the "Category" fieldset legend in the exposed filter block in the left sidebar.
Compare with main branch: https://demo.quickstart.arizona.edu/finders/events

Go to [`/finders/people`](https://9a355b9d-e15c-41ca-a57e-f8b221c9a044.probo.build/finders/people)
Compare with main branch: https://demo.quickstart.arizona.edu/finders/people

Go to [`/finders/news`](https://9a355b9d-e15c-41ca-a57e-f8b221c9a044.probo.build/finders/news)
Compare with main branch: https://demo.quickstart.arizona.edu/finders/news

Go to [`/finders/pages`](https://9a355b9d-e15c-41ca-a57e-f8b221c9a044.probo.build/finders/pages)
Compare with main branch: https://demo.quickstart.arizona.edu/finders/pages

Go to [`/finders/content`](https://9a355b9d-e15c-41ca-a57e-f8b221c9a044.probo.build/finders/content)
Compare with main branch: https://demo.quickstart.arizona.edu/finders/content


It would probably also be worth testing on sites with web forms that have fieldsets with legends in them.


## Screenshots
<img width="1379" alt="image" src="https://github.com/az-digital/az_quickstart/assets/1023167/273140b1-4b11-4e5b-a73b-88890f8c48d9">


## Types of changes


### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal core
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal contrib projects
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

## Checklist
